### PR TITLE
[ParticleMechanicsApplication] Feature hexahedral element 3D8N

### PR DESCRIPTION
**Pull request by bodhinandach**
I extended the UL Quad element to Hexahedral. For the implementation to run, the PR #2386 is needed, else it will throw an error while accessing binbased_fast_point_locator.h's UpdateSearchDatabase() function.

I have validated it with a simple beam example as attached. The exact analytical value is approximately -0.0314
snapshot

### CODE REVIEW

**Review by loumalouomega — CHANGES_REQUESTED** [review]
Wait until #2386 is merged. Besides add that beam as test, please

**Comment by bodhinandach** [comment]
@loumalouomega this one also waiting for approval :bowing_man:

File: applications/ParticleMechanicsApplication/custom_python/particle_mechanics_python_application.cpp
Comment (loumalouomega) [inline applications/ParticleMechanicsApplication/custom_python/particle_mechanics_python_application.cpp]: Use make_shared instead, is more efficient

File: applications/ParticleMechanicsApplication/custom_strategies/strategies/MPM_strategy.h
Comment (loumalouomega) [inline applications/ParticleMechanicsApplication/custom_strategies/strategies/MPM_strategy.h:762]: There are some enums to check the geometry type, please use those ones

**Review by loumalouomega — CHANGES_REQUESTED** [review]
Minor comments

**Comment by bodhinandach** [comment]
ping @loumalouomega

**Review by loumalouomega — APPROVED** [review]
Ok

**Pull request merged**
